build-system/tasks/dist.js

```diff
@@ -438,6 +439,7 @@ dist.flags = {
     '  Outputs compiled code with whitespace. ' +
     'Great for debugging production code.',
   fortesting: '  Compiles production binaries for local testing',
+  no_amp_config: '  Compiles production binaries without applying AMP_CONFIG',
```

Comment:
"no" flags can be confusing, so mycontroversial idea is to rename this flag `prepend_amp_config` with a default value set to `true`; users can then call `gulp dist --prepend_amp_config=false` (or `0` or `no`) to "activate" this flag

Reply:
We have plenty of precedent for "no" flags, like `--noextensions`, `--no_caching_extensions`, `--nobuild`, `--noyarn`, and `--nohelp` (see the output of `gulp help`). In addition, there isn't any precedent for a default `true` flag that needs to be turned off with an `=false`. So I'm inclined to leave this as is.

Reply:
I did notice that almost all but one of our "no" flags contain no underscores (making them easier to type), so I changed the flag to `--noconfig`, to match common use cases like `--noextensions` and `--nobuild`.

/cc @rcebulko, since this will need to be applied to the release script.

Reply:
PS: The only "no" flag with underscores that remains after this is `--no_caching_extensions`, which is an obscure flag for `gulp serve`, and isn't used in automation or in common workflows. I'll leave that as is.